server/src/index.ts

```diff
@@ -3,17 +3,81 @@
 // SPDX-License-Identifier: MIT
 
 import express, { Application, Request, Response } from 'express';
+import { validateLogin, performSignup, PlainPassword, UserRole } from './auth';
 import { TeacherCourses, getTeacherCourses } from './services/courses';
 import cors from 'cors';
 
 const app: Application = express();
 const parsedPort = Number(process.env.AALTO_GRADES_BACKEND_PORT);
 const port: number = isNaN(parsedPort) ? 3000 : parsedPort;
 
+interface LoginRequest {
+  username: string,
+  password: PlainPassword,
+}
+
+interface SignupRequest {
+  username: string,
+  password: PlainPassword,
+  email: string,
+  role: UserRole,
+}
+
+function validateUserRole(role: any): role is UserRole {
+  return typeof role === 'string' && (
+    role === 'Teacher' ||
+    role === 'Student' ||
+    role === 'Admin'
```

Comment:
If these validate functions are to stay, this is seems outdated with changing UserRole to enum.